File: app/src/main/java/me/shreyasr/chatse/event/EventList.java
package me.shreyasr.chatse.event;

import me.shreyasr.chatse.event.presenter.EventPresenter;
import me.shreyasr.chatse.event.presenter.UnfilteredEventPresenter;
import me.shreyasr.chatse.event.presenter.message.MessageEventPresenter;

public class EventList {

    private final int roomNum;

    public UnfilteredEventPresenter unfilteredPresenter = new UnfilteredEventPresenter();
    public MessageEventPresenter messagePresenter = new MessageEventPresenter();

    private EventPresenter[] presenters
            = new EventPresenter[]{unfilteredPresenter, messagePresenter};

    public EventList(int roomNum) {
        this.roomNum = roomNum;
    }

    public void addEvent(ChatEvent event) {
        for (EventPresenter presenter : presenters) {
            presenter.addEvent(event, roomNum);
        }
    }
}

File: app/src/main/java/me/shreyasr/chatse/event/EventList.kt
package me.shreyasr.chatse.event

import android.content.Context
import me.shreyasr.chatse.chat.ChatRoom
import me.shreyasr.chatse.event.presenter.UnfilteredEventPresenter
import me.shreyasr.chatse.event.presenter.message.MessageEventPresenter

/**
 * Maintains a list of events for a specific room.
 */
class EventList(private val roomNum: Int) {

    //List of unfilteredEvents
    var unfilteredPresenter = UnfilteredEventPresenter()
    //List of events used for messages
    var messagePresenter = MessageEventPresenter()

    //Array of the unfiltered and filtered events
    private val presenters = arrayOf(unfilteredPresenter, messagePresenter)

    fun addEvent(event: ChatEvent, context: Context, room: ChatRoom?) {
        presenters.forEach { it.addEvent(event, roomNum, context, room) }
    }
}
